### PR TITLE
Factor out and export the labels printer

### PR DESCRIPTION
**Pull request by lionel-**
This extracts a convenience function to print the labels of a labelled vector. This is useful to explore a newly imported dataset.

### CODE REVIEW

File: R/labelled.R [hunk 1]
```diff
@@ -81,8 +81,33 @@ print.labelled <- function(x, ...) {
...
+#' @examples
+#' \dontrun{
+#' s1 <- labelled(c("M", "M", "F"), c(Male = "M", Female = "F"))
+#' s2 <- labelled(c(1, 1, 2), c(Male = 1, Female = 2))
+#' labelled_df <- dplyr::data_frame(s1, s2)
+#'
+#' labelled_df %>% purrr::walk(print_labels)
+#' labelled_df %>% purrr::walk2(print_labels, names(.))
+#' }
+print_labels <- function(x, name = NULL) {
```
Comment (hadley) [inline R/labelled.R]: This needs to invisibly return it's input (since it's called primarily for its side effect)

File: R/labelled.R [hunk 2]
```diff
@@ -81,8 +81,33 @@ print.labelled <- function(x, ...) {
...
+#' This is a convenience function, useful to explore the variables of
+#' a newly imported dataset.
+#' @param x A labelled vector
+#' @param name The name of the vector (optional)
+#' @export
+#' @examples
+#' \dontrun{
+#' s1 <- labelled(c("M", "M", "F"), c(Male = "M", Female = "F"))
+#' s2 <- labelled(c(1, 1, 2), c(Male = 1, Female = 2))
+#' labelled_df <- dplyr::data_frame(s1, s2)
```
Comment (hadley) [inline R/labelled.R]: Can you switch this to tibble? Haven will probably soon import (#154)

File: R/labelled.R [hunk 3]
```diff
@@ -81,8 +81,33 @@ print.labelled <- function(x, ...) {
...
+#'
+#' labelled_df %>% purrr::walk(print_labels)
```
Comment (hadley) [inline R/labelled.R]: I wish we didn't have to use `purrr` here :/

File: R/labelled.R (hunk 2 above)
Comment (krlmlr) [inline R/labelled.R]: \dontrun{} shouldn't be necessary then, either.

File: R/labelled.R (hunk 3 above)
Comment (lionel-) [inline R/labelled.R]: do you prefer `lapply/Map()` or maybe for loops in these examples?

Comment (hadley) [inline R/labelled.R]: I'd prefer just loops

**Comment by hadley** [comment]
Thanks!

**Pull request merged**